packages/react/src/views/MessageAggregators/SearchMessages.js

```diff
@@ -10,47 +17,99 @@ const SearchMessages = () => {
   const { RCInstance } = useContext(RCContext);
   const [text, setText] = useState('');
   const [messageList, setMessageList] = useState([]);
+  const [error, setError] = useState(null);
+
+  const { theme } = useTheme();
+
+  const requestCount = useRef(0);
+  const lastRequestTime = useRef(0);
+  const requestInterval = 1000;
 
   const handleInputChange = (e) => {
     setText(e.target.value);
+    setError(null);
   };
 
-  const searchMessages = async () => {
-    const { messages } = await RCInstance.getSearchMessages(text);
-    setMessageList(messages);
-  };
+  const searchMessages = useCallback(async () => {
+    const presentTime = Date.now();
+    if (presentTime - lastRequestTime.current < requestInterval) {
```

Comment:
Do we really need to do all this manually ? I don't think so.. 
It looks like a simpler issue, where debounce and rendering has to be controlled that's it

Reply:
Definitely we can use just debounce and callbacks to prevent this !! 
I used the limiter approach to be on a safer side 
Should i go with just using debounce and callbacks  ??